crates/nargo_cli/src/cli/verify_cmd.rs

```diff
@@ -34,15 +34,26 @@ pub(crate) fn run(args: VerifyCommand, config: NargoConfig) -> Result<(), CliErr
         .circuit_name
         .map(|circuit_name| config.program_dir.join(TARGET_DIR).join(circuit_name));
 
-    verify_with_path(config.program_dir, proof_path, circuit_build_path, args.compile_options)
+    let valid_proof = verify_with_path(
+        &config.program_dir,
+        &proof_path,
+        circuit_build_path.as_ref(),
+        args.compile_options,
+    )?;
+
+    if valid_proof {
+        Ok(())
+    } else {
+        Err(CliError::InvalidProof(proof_path))
+    }
 }
 
 fn verify_with_path<P: AsRef<Path>>(
     program_dir: P,
-    proof_path: PathBuf,
+    proof_path: P,
     circuit_build_path: Option<P>,
     compile_options: CompileOptions,
-) -> Result<(), CliError> {
+) -> Result<bool, CliError> {
```

Comment:
Lets move:

```
    if valid_proof {
        Ok(())
    } else {
        Err(CliError::InvalidProof(proof_path))
    }
 ```
 
 code to verify_with_path -- To avoid `Result<bool, Error>` in the function signature since it means that we need to check for the Ok case and then remember to check the boolean too.